tests/Carbon/WeekTest.php

```diff
@@ -13,5305 +13,5312 @@
+        yield ['2019-12-26', [
             2019,
             2019,
             52,
             52,
             52,
             52,
-        ],
-        '2019-12-27' => [
+        ]];
+        yield ['2019-12-27', [
             2019,
             2019,
             52,
             52,
             52,
             52,
-        ],
-        '2019-12-28' => [
+        ]];
+        yield ['2019-12-28', [
             2019,
             2019,
             52,
             52,
             52,
             52,
-        ],
-        '2019-12-29' => [
+        ]];
+        yield ['2019-12-29', [
             2020,
             2019,
             1,
             52,
             52,
             52,
-        ],
-        '2019-12-30' => [
+        ]];
+        yield ['2019-12-30', [
             2020,
             2020,
             1,
             1,
             52,
             52,
-        ],
-        '2019-12-31' => [
+        ]];
+        yield ['2019-12-31', [
             2020,
             2020,
             1,
             1,
             52,
             52,
-        ],
-        '2020-01-01' => [
+        ]];
+        yield ['2020-01-01', [
             2020,
             2020,
             1,
             1,
             52,
             53,
-        ],
-        '2020-01-02' => [
+        ]];
+        yield ['2020-01-02', [
             2020,
             2020,
             1,
             1,
             52,
             53,
-        ],
-        '2020-01-03' => [
+        ]];
+        yield ['2020-01-03', [
             2020,
             2020,
             1,
             1,
             52,
             53,
-        ],
-        '2020-01-04' => [
+        ]];
+        yield ['2020-01-04', [
             2020,
             2020,
             1,
             1,
             52,
             53,
-        ],
-        '2020-01-05' => [
+        ]];
+        yield ['2020-01-05', [
             2020,
             2020,
             2,
             1,
             52,
             53,
-        ],
-        '2020-01-06' => [
+        ]];
+        yield ['2020-01-06', [
             2020,
             2020,
             2,
             2,
             52,
             53,
-        ],
-        '2020-01-07' => [
+        ]];
+        yield ['2020-01-07', [
             2020,
             2020,
             2,
             2,
             52,
             53,
-        ],
-        '2020-01-08' => [
+        ]];
+        yield ['2020-01-08', [
             2020,
             2020,
             2,
             2,
             52,
             53,
-        ],
-        '2020-01-09' => [
+        ]];
+        yield ['2020-01-09', [
             2020,
             2020,
             2,
             2,
             52,
             53,
-        ],
-        '2020-01-10' => [
+        ]];
+        yield ['2020-01-10', [
             2020,
             2020,
             2,
             2,
             52,
             53,
-        ],
-    ];
+        ]];
```

Comment:
For this one, we will have one more tearDown() and setUp() call for each item, so I wonder if it won't actually slow down significantly the test due to the size of the sample.

Reply:
It actually does not, duration for php 8 stable are still around 35-45 seconds with GA
So not much difference

Reply:
And memory wise it will always be more efficient

Reply:
Not true, because PHPUnit immediately convert this Generator into an array. before it re-iterate on it to run tests.